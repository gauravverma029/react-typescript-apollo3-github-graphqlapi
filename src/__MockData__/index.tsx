import { SearchIssues } from "../utils/types/SearchIssues";
import {
  Comments,
  Comments_repository_issue_comments_edges,
} from "../utils/types/Comments";

export const MockDataForIssues: SearchIssues = {
  search: {
    __typename: "SearchResultItemConnection",
    edges: [
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjQ3OTI0ODU=",
          number: 20450,
          state: "OPEN",
          title: "Bug: Trailing Comma in Arrays and Dict",
          createdAt: "2020-12-13T02:33:56Z",
          author: {
            __typename: "User",
            name: "Shankara Narayana",
          },
          labels: {
            __typename: "LabelConnection",
            edges: [
              {
                __typename: "LabelEdge",
                node: {
                  __typename: "Label",
                  name: "Status: Unconfirmed",
                },
              },
            ],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 0,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjM5MDc2NTI=",
          number: 20448,
          state: "OPEN",
          title:
            "[eslint-plugin-react-hooks]: could not find hook called conditionally when using an anonymous function component",
          createdAt: "2020-12-12T14:42:42Z",
          author: {
            __typename: "User",
            name: null,
          },
          labels: {
            __typename: "LabelConnection",
            edges: [
              {
                __typename: "LabelEdge",
                node: {
                  __typename: "Label",
                  name: "Component: ESLint Rules",
                },
              },
            ],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 1,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjM1MTkwMTE=",
          number: 20447,
          state: "OPEN",
          title:
            "Bug: [reactdev tools][react native] Bridge was already shutdown. Inspection is broken",
          createdAt: "2020-12-12T09:48:20Z",
          author: {
            __typename: "User",
            name: null,
          },
          labels: {
            __typename: "LabelConnection",
            edges: [
              {
                __typename: "LabelEdge",
                node: {
                  __typename: "Label",
                  name: "Component: Developer Tools",
                },
              },
            ],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 0,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjMyMTM2OTQ=",
          number: 20446,
          state: "OPEN",
          title: "Using dangerouslySetInnerHTML with <Script> Tag?",
          createdAt: "2020-12-12T02:57:21Z",
          author: {
            __typename: "User",
            name: "chenyang wu",
          },
          labels: {
            __typename: "LabelConnection",
            edges: [],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 1,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjIxNDg3ODk=",
          number: 20435,
          state: "OPEN",
          title:
            "Feature Request(devtools): `launch-editor` for selected component",
          createdAt: "2020-12-11T09:37:45Z",
          author: {
            __typename: "User",
            name: "Eric Chen",
          },
          labels: {
            __typename: "LabelConnection",
            edges: [
              {
                __typename: "LabelEdge",
                node: {
                  __typename: "Label",
                  name: "Component: Developer Tools",
                },
              },
            ],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 1,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjE4NjkwMTQ=",
          number: 20431,
          state: "OPEN",
          title:
            'Error: "Commit tree does not contain fiber 5766. This is a bug in React DevTools."',
          createdAt: "2020-12-11T04:00:11Z",
          author: {
            __typename: "User",
            name: "Dixit Keshavbhai Patel",
          },
          labels: {
            __typename: "LabelConnection",
            edges: [
              {
                __typename: "LabelEdge",
                node: {
                  __typename: "Label",
                  name: "Component: Developer Tools",
                },
              },
            ],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 5,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjE1MjIwMjk=",
          number: 20430,
          state: "CLOSED",
          title:
            'Error: "Commit tree already contains fiber 1719. This is a bug in React DevTools."',
          createdAt: "2020-12-10T19:01:25Z",
          author: {
            __typename: "User",
            name: "Shalom Levi",
          },
          labels: {
            __typename: "LabelConnection",
            edges: [
              {
                __typename: "LabelEdge",
                node: {
                  __typename: "Label",
                  name: "Component: Developer Tools",
                },
              },
            ],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 2,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjA4ODE0NTY=",
          number: 20423,
          state: "CLOSED",
          title: "Bug: ",
          createdAt: "2020-12-10T04:36:53Z",
          author: {
            __typename: "User",
            name: null,
          },
          labels: {
            __typename: "LabelConnection",
            edges: [
              {
                __typename: "LabelEdge",
                node: {
                  __typename: "Label",
                  name: "Status: Unconfirmed",
                },
              },
            ],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 0,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjA2NjkzODA=",
          number: 20418,
          state: "CLOSED",
          title:
            '[React DevTools] Bug: irrelevant "Cannot update a component from inside the function body of a different component"',
          createdAt: "2020-12-09T21:01:14Z",
          author: {
            __typename: "User",
            name: "Jules Sam. Randolph",
          },
          labels: {
            __typename: "LabelConnection",
            edges: [
              {
                __typename: "LabelEdge",
                node: {
                  __typename: "Label",
                  name: "Status: Unconfirmed",
                },
              },
            ],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 1,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
      {
        __typename: "SearchResultItemEdge",
        node: {
          __typename: "Issue",
          id: "MDU6SXNzdWU3NjA2NjEwMDg=",
          number: 20417,
          state: "OPEN",
          title:
            "Bug: react-refresh is refreshing components without signatures and different hooks instead of remount",
          createdAt: "2020-12-09T20:49:59Z",
          author: {
            __typename: "User",
            name: "Bnaya Peretz",
          },
          labels: {
            __typename: "LabelConnection",
            edges: [
              {
                __typename: "LabelEdge",
                node: {
                  __typename: "Label",
                  name: "Component: Fast Refresh",
                },
              },
            ],
          },
          comments: {
            __typename: "IssueCommentConnection",
            totalCount: 2,
          },
          repository: {
            __typename: "Repository",
            nameWithOwner: "facebook/react",
          },
        },
      },
    ],
    issueCount: 9777,
    pageInfo: {
      __typename: "PageInfo",
      endCursor: "Y3Vyc29yOjEw",
      hasNextPage: true,
    },
  },
};

export const MockDataForComments: Comments = {
  repository: {
    __typename: "Repository",
    issue: {
      __typename: "Issue",
      id: "MDU6SXNzdWU3NjMyMTM2OTQ=",
      bodyHTML:
        '<p>Backend return an html type string and I want render in react through</p>\n<div></div>\n<p>however, html contains the JS code that dynamically add data to </p> Tag and it does not render any data in my page<br>\nthe table is empty<p></p><p>How could solve this problem?</p><p>Thanks for any ideas</p><table role="table">\n\n</table>',
      title: "Using dangerouslySetInnerHTML with",
      comments: {
        __typename: "IssueCommentConnection",
        edges: [
          {
            __typename: "IssueCommentEdge",
            node: {
              __typename: "IssueComment",
              id: "MDEyOklzc3VlQ29tbWVudDc0Mzc4MjUxNw==",
              bodyHTML:
                '<p>It doesn\'t look like this bug report has enough info for one of us to reproduce it.</p>\n<p>Please provide a CodeSandbox (<a rel="nofollow" href="https://react.new/">https://react.new/</a>), a link to a repository on GitHub, or provide a minimal code example that reproduces the problem.</p>\n<p>Here are some tips for providing a minimal example: <a rel="nofollow" href="https://stackoverflow.com/help/mcve">https://stackoverflow.com/help/mcve</a></p>\n\n',
              createdAt: "2020-12-12T16:47:43Z",
              author: {
                __typename: "User",
                login: "eps1lon",
              },
            },
          },
        ],
        pageInfo: {
          __typename: "PageInfo",
          endCursor: "Y3Vyc29yOnYyOpHOLFU4dQ==",
          hasNextPage: false,
        },
      },
    },
  },
};

export const MockDataForCommentsNode: Comments_repository_issue_comments_edges = {
  __typename: "IssueCommentEdge",
  node: {
    __typename: "IssueComment",
    id: "MDEyOklzc3VlQ29tbWVudDc0Mzc4MjUxNw==",
    bodyHTML:
      '<p>It doesn\'t look like this bug report has enough info for one of us to reproduce it.</p>\n<p>Please provide a CodeSandbox (<a rel="nofollow" href="https://react.new/">https://react.new/</a>), a link to a repository on GitHub, or provide a minimal code example that reproduces the problem.</p>\n<p>Here are some tips for providing a minimal example: <a rel="nofollow" href="https://stackoverflow.com/help/mcve">https://stackoverflow.com/help/mcve</a></p>\n\n',
    createdAt: "2020-12-12T16:47:43Z",
    author: {
      __typename: "User",
      login: "eps1lon",
    },
  },
};
