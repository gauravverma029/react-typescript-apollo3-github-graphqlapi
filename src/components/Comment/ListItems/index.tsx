import React from "react";
import formatDistance from "date-fns/formatDistance";

import { Comments_repository_issue_comments_edges } from "../../../utils/types/Comments";

type Props = {
  item: Comments_repository_issue_comments_edges | null;
  today: Date;
};

const ListItems: React.FC<Props> = ({ item, today }) => {
  const createdAt = formatDistance(new Date(item?.node?.createdAt), today, {
    addSuffix: true,
  });

  return (
    <article className="hover:bg-gray-200 group block cursor-pointer rounded-md border border-gray-300">
      <div className="px-4 py-3 flex bg-gray-100 text-lg font-semibold text-black mb-0.5 break-all">
        {`${item?.node?.author?.login}  ${createdAt}`}
      </div>
      <div className="min-w-0 p-4 relative flex-auto  break-all">
        <div dangerouslySetInnerHTML={{ __html: item?.node?.bodyHTML }} />
      </div>
    </article>
  );
};
export default ListItems;
