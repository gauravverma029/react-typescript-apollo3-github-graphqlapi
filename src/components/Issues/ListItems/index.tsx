import React from "react";
import formatDistance from "date-fns/formatDistance";
import { Link } from "react-router-dom";
import { SearchIssues_search_edges } from "../../../utils/types/SearchIssues";

type Props = {
  item: SearchIssues_search_edges | null;
  today: Date;
};

const ListItems: React.FC<Props> = ({ item, today }) => {
  const createdAt = formatDistance(new Date(item?.node?.createdAt), today, {
    addSuffix: true,
  });

  return (
    <div className="border-b-1 border-gray-400 hover:bg-gray-200 group block p-4 cursor-pointer">
      <Link
        to={`/${item?.node?.number}`}
        className="divide-y-2 divide-gray-400"
      >
        <div className="flex flex-col justify-between lg:space-x-32 md:flex-row md:space-x-4">
          <div className="flex-2">
            <div className="flex flex-col">
              <div className="text-md font-semibold text-black mb-0.5 break-all lg:text-md">
                {`${item?.node?.repository?.nameWithOwner} - ${item?.node?.title}`}
              </div>
              <div className="text-gray-400 font-light flex flex-wrap text-sm whitespace-pre truncate md:overflow-clip">
                {`#${item?.node?.number} ${createdAt}`}
              </div>
            </div>
          </div>

          <div className="flex flex-row py-2 lg:py-0.5">
            <div className="text-gray-600 p-2 mx-2 h-9 flex items-center justify-center bg-gray-100 rounded-lg">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                fill="none"
                viewBox="0 0 24 24"
                stroke="currentColor"
                width="20"
                height="20"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                  d="M8 10h.01M12 10h.01M16 10h.01M9 16H5a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v8a2 2 0 01-2 2h-5l-5 5v-5z"
                />
              </svg>
              <div>{item?.node?.comments.totalCount}</div>
            </div>
            <div className="bg-gray-100 text-gray-600 w-20 p-2 mx-2 h-9 flex items-center justify-center rounded-lg">
              {item?.node?.state}
            </div>
          </div>
        </div>
      </Link>
    </div>
  );
};
export default ListItems;
