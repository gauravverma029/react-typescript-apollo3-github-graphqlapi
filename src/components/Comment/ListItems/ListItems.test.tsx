import React from "react";
import { cleanup, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import { MockDataForCommentsNode } from "../../../__MockData__";

import ListItems from "./index";

afterEach(cleanup);

describe("Comments ListItems Component", () => {
  const today = new Date("2020-12-13T02:33:56Z");

  test("Should Render and mock data", async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <BrowserRouter>
          <ListItems item={MockDataForCommentsNode} today={today} />
        </BrowserRouter>
      </BrowserRouter>
    );

    const html = asFragment();
    expect(html).toMatchSnapshot();
  });

  test("Should Render title", async () => {
    const { getByText } = render(
      <BrowserRouter>
        <ListItems item={MockDataForCommentsNode} today={today} />
      </BrowserRouter>
    );
    getByText(/look like this bug report/i);
  });
});
