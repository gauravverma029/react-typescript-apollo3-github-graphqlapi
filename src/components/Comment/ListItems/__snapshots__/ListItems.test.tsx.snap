// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments ListItems Component Should Render and mock data 1`] = `
<DocumentFragment>
  <article
    class="hover:bg-gray-200 group block cursor-pointer rounded-md border border-gray-300"
  >
    <div
      class="px-4 py-3 flex bg-gray-100 text-lg font-semibold text-black mb-0.5 break-all"
    >
      eps1lon  about 10 hours ago
    </div>
    <div
      class="min-w-0 p-4 relative flex-auto  break-all"
    >
      <div>
        <p>
          It doesn't look like this bug report has enough info for one of us to reproduce it.
        </p>
        

        <p>
          Please provide a CodeSandbox (
          <a
            href="https://react.new/"
            rel="nofollow"
          >
            https://react.new/
          </a>
          ), a link to a repository on GitHub, or provide a minimal code example that reproduces the problem.
        </p>
        

        <p>
          Here are some tips for providing a minimal example: 
          <a
            href="https://stackoverflow.com/help/mcve"
            rel="nofollow"
          >
            https://stackoverflow.com/help/mcve
          </a>
        </p>
        


      </div>
    </div>
  </article>
</DocumentFragment>
`;
